### ISSUE TITLE
How to query by FieldPath.documentId()?

### ISSUE THREAD

**Issue by reedom** [kossnocorp/typesaurus#63]
Is there equivalent feature of `FieldPath.documentId()`?

I was tried to query "documentId-starts with",
if there was the one, it'd look like:

```js
query(items, [
  where(documentId(), ">=", "abc"),
  where(documentId(), "<", "abd")
])
```

**Comment by kossnocorp** [comment]
Hi! Sorry, there's none at the moment as I didn't know it exists!

I can add it the next week, or gladly accept a pull-request. In case if you want to help, I'd like to name it `docId`.

**Comment by reedom** [comment]
I wanted to be a help so that I had tacked for hours.
But, no, Typescript's type system is too complicated to me yet... sorry 😭😭😭

**Comment by reedom** [comment]
Only the test code snippet I wrote this time might be reusable as a draft.
https://github.com/reedom/typesaurus/blob/65b9d24d43a297ac66593cdf78f88fcceb71105c/src/query/test.ts#L487..L540

**Comment by kossnocorp** [comment]
No worries, I'll tackle that! Thank you a lot for the effort. I appreciate it!

**Comment by kossnocorp** [comment]
Thank you so much, I finished the work you started and published it with v6.2.0!

**Comment by reedom** [comment]
Thanks for adopting the `docId` feature itself.

But..., it looks as that my patch code for the main part has also adopted, this is unexpected to me 😨 

As I wrote
> Only the test code snippet I wrote this time might be reusable as a draft.

The reason why I implied not to use the patch code other than test code was that it had broke type restriction
of typesaurus' some features; dismissed one of _raison d'être_ of the library.

For example,
https://github.com/kossnocorp/typesaurus/blob/master/src/query/test.ts#L455
```typescript
      const docs = await query(contacts, [
        where('ownerId', '==', ownerId),
        order('year', 'asc', [startAt(1989), endAt(1989)]),
```

`startAt()` used to accept only `numeric` values while the `year` field is `numeric` typed,
thanks for the type inference.

With my code, now `tsc` misses `string` in many places, e.g.
```typescript
        order('year', 'asc', [startAt("1989"), endAt(1989)]),  // tcs become won't blame against this line...
```

I'm glad enabling `docId` query feature since I've used many places, though,
I wish the library regains the maximum type inference ability.
For that, to revert the docId feature could be an option, I think.
(I had tried to solve this issue even days after sending the patch, but still was too hard for me...)
